waspc/src/Wasp/Util.hs

```diff
@@ -57,6 +58,13 @@ camelToKebabCase camel@(camelHead : camelTail) = kebabHead : kebabTail
         (zip camel camelTail)
     isCamelHump (a, b) = (not . isUpper) a && isUpper b
 
+projectNameToAppIdentifier :: String -> String
+projectNameToAppIdentifier = concat . capitalizeWords . wordsBy (== '-')
+  where
+    capitalizeWords :: [String] -> [String]
+    capitalizeWords [] = []
+    capitalizeWords (kebabHead : kebabTail) = kebabHead : map toUpperFirst kebabTail
```

Comment:
True, too specific! I would actually move it next to definition of `ProjectName` data type, and I think that solves the problem.

Taht said, you could also extract the function `kebabToCamelCase` and put that one here in `Util.hs`. And then `projectNameToAppIdentifier` would just call this one, for now. But as I mentioned above it would be moved out of Util.hs.

So `kebabToCamelCase` in Util.hs, `projectNameToWaspIdentifier` in `CreateNewProject.hs`.

Reply:
I'll write a conclusion of our comments because there are many similar terms (project name, app identifier, wasp identifier...) flying around, which has become confusing.

There are three relevant terms here:
- _Project name_ - The name of the project (i.e., directory)
- _App name_ - The name of the Wasp application (written inside the .wasp file)
- _Wasp identifier_ - Any string that can serve as a valid identifier inside the Wasp language

At the moment, _Project name_ and _App name_ are the same thing, and they must be a _Wasp identifier_. We want to relax this a bit by allowing kebab-case for _Project names_. However, we must then derive the _App name_ from the _Project name_ because the _App name_ must still be a _Wasp identifier_.

These are the changes we should make:
- This function (`projectNameToAppIdentifier`) should be renamed to `kebabToCamelCase`
- A  _Project name_ is valid only if we can turn it into a _Wasp identifier_. Therefore, we should remove the types `AppIdentifier` and `ProjectName` and only implement the following: 
```haskell
data ProjectInfo = ProjectInfo {
  _projectName :: String,
  _appName :: String
}

-- Returns either the ProjectInfo type that contains both the Project name 
-- and the App name (which might be the same), or an error describing why the name is invalid
parseProjectInfo :: String -> Either String ProjectInfo
parseProjectInfo = -- Internally uses the functions 'kebabToCamelCase' and 'isValidWaspIdentifier'
```
- The rest of the code should then use the created `ProjectInfo` object which contains everything anyone might need.

Reply:
Thank you both. I tried to implement from what I gathered & from the summary.
Let me know what you think